modules/api/src/main/java/org/apache/ignite/tx/IgniteTransactions.java

```diff
@@ -130,4 +132,22 @@ default <T> T runInTransaction(Function<Transaction, T> clo) throws TransactionE
             throw t;
         }
     }
+
+    /**
+     * Executes a closure within a transaction asynchronously.
+     *
+     * <p>A returned future must be the last in the asynchronous chain. This means all transaction operations happen before the future
+     * is completed.
+     *
+     * <p>If the asynchronous chain resulted in no exception, the commitAsync will be automatically called.
+     *
+     * @param clo The closure.
+     * @param <T> Closure result type.
+     * @return The result.
+     */
+    default <T> CompletableFuture<T> runInTransactionAsync(Function<Transaction, CompletableFuture<T>> clo) {
+        // Rollback is expected to be called by the failure handling code
```

Comment:
What do you mean?
It seems as inconsistency to me. As far as I understand you expect user to have op.handle(tx.rollback) within closure despite the fact that sync version of runInTransaction will rollback the tx by her own.
```
    default <T> T runInTransaction(Function<Transaction, T> clo) throws TransactionException {
        Transaction tx = begin();

        try {
            T ret = clo.apply(tx);

            System.out.println("!!!: Before commit");
            tx.commit();

            return ret;
        } catch (Throwable t) {
            try {
                tx.rollback(); // Try rolling back on user exception.
            } catch (Exception e) {
                t.addSuppressed(e);
            }

            throw t;
        }
    }
```

Reply:
**As far as I understand you expect user to have op.handle(tx.rollback)** - no I don't expect this. 
I expect that a transaction will be already rolled back by a user or automatically by txn engine (in case of error) by the async chain end, so there is no need to call it again.

Reply:
It's still possible that an exception will be thrown not from async operations, please consider following scenario.
```
    @Test
    public void testTxClosureKeyValueView2() throws TransactionException {
        var f = igniteTransactions.runInTransactionAsync(tx -> {
            KeyValueView<Tuple, Tuple> view = accounts.keyValueView();

            CompletableFuture<Tuple> read1 = view.getAsync(tx, makeKey(1));

            System.out.println(10 / 0);

            return read1;
        });

        f.join();
    }
```

Reply:
Catch block and test added, javadoc is improved